unstructured/ingest/v2/processes/connectors/azure_cognitive_search.py

```diff
@@ -0,0 +1,225 @@
+from unstructured.ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+    add_destination_entry,
+)
+from unstructured.utils import requires_dependencies
+
+if t.TYPE_CHECKING:
+    from azure.search.documents import SearchClient
+
+
+CONNECTOR_TYPE = "azure_cognitive_search"
+
+
+@dataclass
+class AzureCognitiveSearchAccessConfig(AccessConfig):
+    key: t.Optional[str] = enhanced_field(default=None, overload_name="azure_cognitive_search_key")
+
+
+@dataclass
+class AzureCognitiveSearchConnectionConfig(ConnectionConfig):
+    endpoint: str
+    index: str
+    access_config: AzureCognitiveSearchAccessConfig = enhanced_field(sensitive=True)
+
+    @requires_dependencies(["azure.search", "azure.core"], extras="azure-cognitive-search")
+    def generate_client(self) -> "SearchClient":
+        from azure.core.credentials import AzureKeyCredential
+        from azure.search.documents import SearchClient
+
+        return SearchClient(
+            endpoint=self.endpoint,
+            index_name=self.index,
+            credential=AzureKeyCredential(self.access_config.key),
+        )
+
+
+@dataclass
+class AzureCognitiveSearchUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class AzureCognitiveSearchUploaderConfig(UploaderConfig):
+    batch_size: int = 100
+    num_processes: int = 1
+
+
+@dataclass
+class AzureCognitiveSearchUploadStager(UploadStager):
+    upload_stager_config: AzureCognitiveSearchUploadStagerConfig = field(
+        default_factory=lambda: AzureCognitiveSearchUploadStagerConfig()
+    )
+
+    @staticmethod
+    def conform_dict(data: dict) -> dict:
+        """
+        updates the dictionary that is from each Element being converted into a dict/json
+        into a dictionary that conforms to the schema expected by the
+        Azure Cognitive Search index
+        """
+        from dateutil import parser
+
+        data["id"] = str(uuid.uuid4())
+
+        if points := data.get("metadata", {}).get("coordinates", {}).get("points"):
+            data["metadata"]["coordinates"]["points"] = json.dumps(points)
+        if version := data.get("metadata", {}).get("data_source", {}).get("version"):
+            data["metadata"]["data_source"]["version"] = str(version)
+        if record_locator := data.get("metadata", {}).get("data_source", {}).get("record_locator"):
+            data["metadata"]["data_source"]["record_locator"] = json.dumps(record_locator)
+        if permissions_data := (
+            data.get("metadata", {}).get("data_source", {}).get("permissions_data")
+        ):
+            data["metadata"]["data_source"]["permissions_data"] = json.dumps(permissions_data)
+        if links := data.get("metadata", {}).get("links"):
+            data["metadata"]["links"] = [json.dumps(link) for link in links]
+        if last_modified := data.get("metadata", {}).get("last_modified"):
+            data["metadata"]["last_modified"] = parser.parse(
```

Comment:
I feel like these sets of datetime formats are a bit awkward.

The ensure_isoformat_datetime potential uses the dateutil parse as well, then we convert it back to string and again parse.

Maybe once this PR is merged: https://github.com/Unstructured-IO/unstructured/pull/3314/files#diff-903a1fa78fa219cd00b0ca836301f9bcaeef3db6e57e30606f687e36bccf17eb

You can refactor these date handlers to something like `parse_datetime(last_modified).strftime("%Y-%m-%dT%H:%M:%S.%fZ")`

Reply:
These should be addressing a bunch of our review points! (and some unspoken ones)

- [remove multiprocessing](https://github.com/Unstructured-IO/unstructured/pull/3311/commits/c67307f9c57ec3d5b01cc0b8ec9906c8bd7f441a)
- [duplicate date parsing logic within pull](https://github.com/Unstructured-IO/unstructured/pull/3311/commits/6743634507c6a362b247d8c440eebac272135a16)
- [revert added date parsing logic within string_and_date_utils](https://github.com/Unstructured-IO/unstructured/pull/3311/commits/9cfd0669c77ff54a8f821aa44beb06b1dc1cc244)
- [update batching function name](https://github.com/Unstructured-IO/unstructured/pull/3311/commits/543a76c3346eb3d3f279eebdcf678b6f116eb307)